Revamp of findings format at Slack and Teams

kbhagi opened the pull request:
## List of files changed
add-ons/electriceye-chatops/cloudformation/ElectricEye_ChatOps_CFN.yml
add-ons/electriceye-chatops/microsoft-teams/cloudformation/ElectricEye_ChatOps_Teams_CFN.yml
add-ons/electriceye-chatops/raw-source/ElectricEye-ChatOps.py
add-ons/electriceye-chatops/terraform/lambda_function.zip
add-ons/electriceye-chatops/terraform/main.tf
add-ons/electriceye-chatops/terraform/variables.tf

* Added a new parameter SSMParameterName at add-ons/electriceye-chatops/cloudformation/ElectricEye_ChatOps_CFN.yml
* Environment Variable named changed to SSM_PARAMETER_NAME
* Raw source code updated at add-ons/electriceye-chatops/raw-source/ElectricEye-ChatOps.py

#### SSM Parameter Store format required for Slack :-
```
{
   "bot_token" : "xxxx",
   "slack_channel_id"  :  "XXXXXXXXX"
}
```


* Added a new parameter SSMParameterNameForTeamsWebhook at add-ons/electriceye-chatops/microsoft-teams/cloudformation/ElectricEye_ChatOps_Teams_CFN.yml
* Environment Variable named changed to SSM_PARAMETER_NAME at add-ons/electriceye-chatops/microsoft-teams/cloudformation/ElectricEye_ChatOps_Teams_CFN.yml


#### SSM Parameter Store format required for Teams  :-
```
{
   "teams_webhook"  :  "XXXXXXXXX"
}
```

### Terraform changes :-
* Raw source code updated at add-ons/electriceye-chatops/terraform/lambda_function.zip
* variable name updated to SSM_PARAMETER_NAME at add-ons/electriceye-chatops/terraform/main.tf
* variable name updated to SSM_PARAMETER_NAME at add-ons/electriceye-chatops/terraform/variables.tf

jonrau1 commented:
@kbhagi LGTM just need to add back in the Lambda Layer plus TF Var/CFN Parameter for a Layer ARN (preferably from Klayers) since the switch was made from urllib3

kbhagi commented:
I understand that request module is only available natively at python 3.7 runtime of AWS Lambda. Rest of the runtimes such as python 3.8, python 3.9 and python 3.6 do not contain requests. However, all the runtime consist of urllib3. This was the actual reason for using urllib3. However, using this urllib3 I was never able to make a successfull HTTP POST request to the new slack api or the classic Teams webhook. [1]

Using a requests seemed pretty easy since this was wrapper around the urllib3. Ideally, I felt that using the official slack sdk [2] is right. As you have suggested, using the Klayers was the simplest way to make-up for the non-existing packages

### Changes done :-

-   Added parameter RequestsPackageKlayersArn and its reference as Layer for Lambda resource at ElectricEye\add-ons\electriceye-chatops\cloudformation\ElectricEye_ChatOps_CFN.yml

-  Added parameter RequestsPackageKlayersArn and its reference as Layer for Lambda resource at ElectricEye\add-ons\electriceye-chatops\microsoft-teams\cloudformation\ElectricEye_ChatOps_Teams_CFN.yml

-  Added a layer resource at ElectricEye\add-ons\electriceye-chatops\terraform\main.tf

- Added a variable at ElectricEye\add-ons\electriceye-chatops\terraform\variables.tf

-  Updated SSMParameterName at ElectricEye\add-ons\electriceye-chatops\terraform\variables.tf to match the environment variable reference at main.tf

References:
[1] https://api.slack.com/
[2] https://github.com/slackapi/python-slack-sdk
